packages/neos-ui-editors/src/Editors/AssetEditor/index.js

```diff
@@ -192,85 +192,92 @@ export default class AssetEditor extends PureComponent {
         );
     }
 
-    renderAssetUpload() {
-        if (!this.isFeatureEnabled('upload')) {
```

Comment:
This is the root of the problem, so no upload, editor is basically empty. You cannot remove an asset ever again.